autonat.go

```diff
@@ -336,24 +391,28 @@ func (as *AmbientAutoNAT) probeNextPeer() {
 			continue
 		}
 
+		// Exclude peers in backoff.
+		if lastTime, ok := as.recentProbes[p]; ok {
+			if time.Since(lastTime) < as.throttlePeerPeriod {
+				continue
+			}
+		}
+
 		if !as.config.dialPolicy.skipPeer(info.Addrs) {
-			addrs = append(addrs, info)
+			candidates = append(candidates, p)
 		}
```

Comment:
nit: use `if ... { continue }` for all checks, then append at the end. It makes this kind of logic a bit easier to read.